src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -129,7 +130,8 @@ public RoutingResponse route() {
       request.preferences().system().itineraryFilters(),
       request.numItineraries(),
       filterOnLatestDepartureTime(),
-      emptyDirectModeHandler.removeWalkAllTheWayResults(),
+      emptyDirectModeHandler.removeWalkAllTheWayResults() ||
+      request.journey().direct().mode() == StreetMode.FLEXIBLE,
```

Comment:
Can you move this into a method or a local variable so the intention become a bit more clear?

Reply:
Extract expression `request.journey().direct().mode() == StreetMode.FLEXIBLE` and give it a proper name